src/main/java/Duke.java

```diff
@@ -1,10 +1,202 @@
+        System.out.println(BAR);
+        System.out.println("OK, I've marked this task as not done yet:");
+        printTask(curr);
+        System.out.println(BAR);
+    }
+
+    private void handleTodo(String[] inputArray, String originalInput) {
+        try {
+            if (inputArray.length <= 1) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            }
+        } catch (DukeException e) {
+            System.err.println(e.getMessage());
+            System.out.println("Please try again:");
+            return;
+        }
+
+        System.out.println(BAR);
+        System.out.println("Got it. I've added this task:");
+        Task curr = new Todo(originalInput.replaceAll(".*\\btodo\\.*", ""));
+        processNewTask(curr);
+        System.out.println(BAR);
+    }
+
+    private void handleDeadline(String originalInput) {
+        String metaInfo = originalInput.split("/by")[1];
+        String strippedCommand = originalInput.replaceAll(".*\\bdeadline\\.*", "");
+        System.out.println(BAR);
+        System.out.println("Got it. I've added this task:");
+        Task curr = new Deadline(strippedCommand.split("/")[0], metaInfo);
+        processNewTask(curr);
+        System.out.println(BAR);
+    }
+
+    private void handleEvent(String originalInput) {
+        String metaInfo = originalInput.split("/at")[1];
+        String strippedCommand = originalInput.replaceAll(".*\\bevent\\.*", "");
+        System.out.println(BAR);
+        System.out.println("Got it. I've added this task:");
+        Task curr = new Event(strippedCommand.split("/")[0], metaInfo);
+        processNewTask(curr);
+        System.out.println(BAR);
+    }
+
+    private void handleDelete(String[] inputArray) {
+        int number = Integer.parseInt(inputArray[1]);
+
+        try {
+            if (((number) <= 0) || ((number) > tasks.size())) {
+                throw new DukeException("Hey! That item does not exist!");
+            }
+        } catch (DukeException e) {
+            System.err.println(e.getMessage());
+            System.out.println("Please try again:");
+            return;
+        }
+
+        Task curr = tasks.get(number - 1);
+        String message = curr.getTaskIcon() + " [" + curr.isDone() + "]" + curr;
+        tasks.remove(curr);
+
+        System.out.println(BAR);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(message);
+        printNoOfTasks();
+        System.out.println(BAR);
+    }
+
+    /**
+     * Main logic for the chatbot
+     * Gets called in the main method
+     * 
+     * @throws DukeException if user input is invalid
+     */
+    private void run() {
+        Scanner sc = new Scanner(System.in);
+
+        String greeting = "Hello! I'm Kizer\nWhat can I do for you?";
+        System.out.println(BAR);
+        System.out.println(greeting);
+        System.out.println(BAR);
+
+        boolean canContinue = true;
```

Comment:
You can consider not using this variable canContinue, but rather return from the function if the user input for the "bye" is entered